Show meaningful error when routes aren't defined. Fixes #80.

msausa opened the pull request:
Signed-off-by: Michelle Sausa <michelle@rightscale.com>

msausa commented:
Fixed to raise an error which specifies the action and resource for missing routes when running `rake praxis:routes` but does not raise anything with `rake praxis:api_docs`. It will still generate the doc with the route missing. Do we also want this to show an error?

careo commented on lib/praxis/tasks/routes.rb:
```diff
@@ -22,6 +21,10 @@
+        if action.routes.empty?
+          raise "No routes defined for #{resource_definition.name}##{name}."
+        end
+
```
I think rather than raise here, we should add a row for the action that just has no route information. 

Something like this if the routes are empty:
```ruby
rows << {
  resource: resource_definition.name,
  version: '',
  verb: '',
  path: '',
  action: name,
  implementation: method_name,
  name: '',
  primary: ''
}
```

msausa commented:
Updated to reflect Dane's suggestions to allow an action to have an undefined route. Added a warning message which calls out missing route for a given resource definition.

careo commented:
LGTM.
